# Data Cleaning & Manipulating a TSV file Using Pandas

In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('chipotle.tsv', delimiter='\t')
data

order_id  quantity                              item_name   
0            1         1           Chips and Fresh Tomato Salsa  \
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]     $3.39   
2                                               [Apple]     $3.39   
3                                                   NaN     $2.39   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
...                                                 ...        ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75   

[4622 rows x 5 columns]

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


## Deleting Duplicate Rows

In [4]:
data.duplicated().sum()

59

In [5]:
df1 = data.copy()

In [6]:
df1.drop_duplicates(inplace=True)
df1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 4563 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4563 non-null   int64 
 1   quantity            4563 non-null   int64 
 2   item_name           4563 non-null   object
 3   choice_description  3335 non-null   object
 4   item_price          4563 non-null   object
dtypes: int64(2), object(3)
memory usage: 213.9+ KB


## Converting item_price data type


In [7]:
df2 = df1.copy()

In [8]:
df2['item_price'] = df2['item_price'].str.replace('$', '').astype(float)
df2.head()

order_id  quantity                              item_name   
0         1         1           Chips and Fresh Tomato Salsa  \
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description  item_price  
0                                                NaN        2.39  
1                                       [Clementine]        3.39  
2                                            [Apple]        3.39  
3                                                NaN        2.39  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98

In [9]:
df2['item_price'].dtype

dtype('float64')

## Investigating missing values

In [10]:
df3 = df2.copy()

In [11]:
df3.isnull().sum()

order_id                 0
quantity                 0
item_name                0
choice_description    1228
item_price               0
dtype: int64

In [12]:
#percentage of missing values

(df3.isnull().sum() / len(df3)) * 100

order_id               0.000000
quantity               0.000000
item_name              0.000000
choice_description    26.912119
item_price             0.000000
dtype: float64

In [13]:
null_rows = df3[df3.isnull().any(axis=1)]
null_rows

order_id  quantity                              item_name   
0            1         1           Chips and Fresh Tomato Salsa  \
3            1         1  Chips and Tomatillo-Green Chili Salsa   
6            3         1                          Side of Chips   
10           5         1                    Chips and Guacamole   
14           7         1                    Chips and Guacamole   
...        ...       ...                                    ...   
4600      1827         1                    Chips and Guacamole   
4605      1828         1                    Chips and Guacamole   
4613      1831         1                                  Chips   
4614      1831         1                          Bottled Water   
4616      1832         1                    Chips and Guacamole   

     choice_description  item_price  
0                   NaN        2.39  
3                   NaN        2.39  
6                   NaN        1.69  
10                  NaN        4.45  
14                  NaN        4.45  
...                 ...         ...  
4600                NaN        4.45  
4605                NaN        4.45  
4613                NaN        2.15  
4614                NaN        1.50  
4616                NaN        4.45  

[1228 rows x 5 columns]

In [14]:
null_rows['item_name'].value_counts()

item_name
Chips and Guacamole                      474
Chips                                    208
Bottled Water                            155
Chips and Fresh Tomato Salsa             110
Side of Chips                            101
Chips and Tomatillo Red Chili Salsa       46
Chips and Tomatillo Green Chili Salsa     43
Chips and Tomatillo-Green Chili Salsa     31
Chips and Roasted Chili Corn Salsa        22
Chips and Tomatillo-Red Chili Salsa       19
Chips and Roasted Chili-Corn Salsa        18
Chips and Mild Fresh Tomato Salsa          1
Name: count, dtype: int64

As seen from the investigation above, the null values in the `choice_description` column correspond to the items that do not have customisation options such as Bottled Water, Side of Chips etc. <br><br>
The above data in concern takes up 26.9% of space in the entire dataset and shows unique values that cannot be removed or filled arbitarily with measures of central tendencies. Therefore, it is in the best interest of the data analyst to leave the null cells as it is. 

## Quantity and Price columns

In [16]:
df4 = df3.copy()

In [17]:
df4['quantity'].value_counts()

quantity
1     4296
2      224
3       28
4       10
5        1
15       1
7        1
8        1
10       1
Name: count, dtype: int64

In [18]:
#testing if any item_price is in negative values

df4[df4['item_price'] < 0]

Empty DataFrame
Columns: [order_id, quantity, item_name, choice_description, item_price]
Index: []

In [19]:
df4[df4['quantity'] > 4]

order_id  quantity                     item_name choice_description   
2441       970         5                 Bottled Water                NaN  \
3598      1443        15  Chips and Fresh Tomato Salsa                NaN   
3599      1443         7                 Bottled Water                NaN   
3887      1559         8                 Side of Chips                NaN   
4152      1660        10                 Bottled Water                NaN   

      item_price  
2441        7.50  
3598       44.25  
3599       10.50  
3887       13.52  
4152       15.00

In [20]:
#testing relation b/w quantity and item price
# 1.69 is the price of one side of chips
#row index 3887 shows 8 quantities of side of chips sold for 13.52

1.69*8

        #according to the result, we see that the item price = price of one * quantity 

13.52

## Choice Description Column
The choice description column contains nested lists of additional toppings added to each item in the order.

In [91]:
df5 = df4.copy()

In [94]:
#counting the number of items added

choice_count = df5['choice_description'].str.split(',').str.len()
choice_count

0       NaN
1       1.0
2       1.0
3       NaN
4       5.0
       ... 
4617    7.0
4618    6.0
4619    5.0
4620    3.0
4621    4.0
Name: choice_description, Length: 4563, dtype: float64

In [95]:
df5.columns

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

In [96]:
#inserting the count of added items in the df

choice_count_index = 4

df5.insert(choice_count_index, 'choice_count', choice_count)

df5

order_id  quantity                              item_name   
0            1         1           Chips and Fresh Tomato Salsa  \
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description  choice_count   
0                                                   NaN           NaN  \
1                                          [Clementine]           1.0   
2                                               [Apple]           1.0   
3                                                   NaN           NaN   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...           5.0   
...                                                 ...           ...   
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...           7.0   
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...           6.0   
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...           5.0   
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...           3.0   
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...           4.0   

      item_price  
0           2.39  
1           3.39  
2           3.39  
3           2.39  
4          16.98  
...          ...  
4617       11.75  
4618       11.75  
4619       11.25  
4620        8.75  
4621        8.75  

[4563 rows x 6 columns]

## OrderID Column

The order_id column shows multiple row entries of the same order id. <br> <br>
To assist analysts, we will create a new df that contains aggregate values grouped by  the order_id

In [104]:
orderid_df = df5.groupby('order_id').agg({'quantity': 'sum', 'choice_count': 'sum','item_price': 'sum'}).reset_index()

orderid_df

order_id  quantity  choice_count  item_price
0            1         4           2.0       11.56
1            2         2           5.0       16.98
2            3         2           6.0       12.67
3            4         2          13.0       21.00
4            5         2           7.0       13.70
...        ...       ...           ...         ...
1829      1830         2          11.0       23.00
1830      1831         3           7.0       12.90
1831      1832         2           4.0       13.20
1832      1833         2          13.0       23.50
1833      1834         3          12.0       28.75

[1834 rows x 4 columns]

## Item Name Column

We will list all the unique items ordered in this dataset and add an item_code to it.

In [131]:
df6 = df5.copy()

In [133]:
items = df6['item_name'].value_counts().sort_index(ascending=True)
items

item_name
6 Pack Soft Drink                         54
Barbacoa Bowl                             65
Barbacoa Burrito                          90
Barbacoa Crispy Tacos                     11
Barbacoa Salad Bowl                        9
Barbacoa Soft Tacos                       25
Bottled Water                            155
Bowl                                       2
Burrito                                    6
Canned Soda                              102
Canned Soft Drink                        290
Carnitas Bowl                             68
Carnitas Burrito                          59
Carnitas Crispy Tacos                      7
Carnitas Salad                             1
Carnitas Salad Bowl                        6
Carnitas Soft Tacos                       40
Chicken Bowl                             717
Chicken Burrito                          546
Chicken Crispy Tacos                      47
Chicken Salad                              9
Chicken Salad Bowl                       110


In [134]:
#converting the results into a dataframe

item_df = pd.DataFrame({'item_name': items.index, 'order_count': items.values})
item_df.head()

item_name  order_count
0      6 Pack Soft Drink           54
1          Barbacoa Bowl           65
2       Barbacoa Burrito           90
3  Barbacoa Crispy Tacos           11
4    Barbacoa Salad Bowl            9

In [135]:
#adding item code on the item_name arranged in an asc order

item_df['item_code'] = range(len(item_df))

item_df

item_name  order_count  item_code
0                       6 Pack Soft Drink           54          0
1                           Barbacoa Bowl           65          1
2                        Barbacoa Burrito           90          2
3                   Barbacoa Crispy Tacos           11          3
4                     Barbacoa Salad Bowl            9          4
5                     Barbacoa Soft Tacos           25          5
6                           Bottled Water          155          6
7                                    Bowl            2          7
8                                 Burrito            6          8
9                             Canned Soda          102          9
10                      Canned Soft Drink          290         10
11                          Carnitas Bowl           68         11
12                       Carnitas Burrito           59         12
13                  Carnitas Crispy Tacos            7         13
14                         Carnitas Salad            1         14
15                    Carnitas Salad Bowl            6         15
16                    Carnitas Soft Tacos           40         16
17                           Chicken Bowl          717         17
18                        Chicken Burrito          546         18
19                   Chicken Crispy Tacos           47         19
20                          Chicken Salad            9         20
21                     Chicken Salad Bowl          110         21
22                     Chicken Soft Tacos          111         22
23                                  Chips          208         23
24           Chips and Fresh Tomato Salsa          110         24
25                    Chips and Guacamole          474         25
26      Chips and Mild Fresh Tomato Salsa            1         26
27     Chips and Roasted Chili Corn Salsa           22         27
28     Chips and Roasted Chili-Corn Salsa           18         28
29  Chips and Tomatillo Green Chili Salsa           43         29
30    Chips and Tomatillo Red Chili Salsa           46         30
31  Chips and Tomatillo-Green Chili Salsa           31         31
32    Chips and Tomatillo-Red Chili Salsa           19         32
33                           Crispy Tacos            2         33
34                                   Izze           19         34
35                       Nantucket Nectar           27         35
36                                  Salad            2         36
37                          Side of Chips          101         37
38                             Steak Bowl          210         38
39                          Steak Burrito          365         39
40                     Steak Crispy Tacos           35         40
41                            Steak Salad            4         41
42                       Steak Salad Bowl           29         42
43                       Steak Soft Tacos           55         43
44                            Veggie Bowl           85         44
45                         Veggie Burrito           95         45
46                    Veggie Crispy Tacos            1         46
47                           Veggie Salad            6         47
48                      Veggie Salad Bowl           18         48
49                      Veggie Soft Tacos            7         49

In [136]:
#creating a merged dataframe to combine the item code column with the main df

merged_df = pd.merge(df6, item_df, on='item_name', how='left')
merged_df

order_id  quantity                              item_name   
0            1         1           Chips and Fresh Tomato Salsa  \
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4558      1833         1                          Steak Burrito   
4559      1833         1                          Steak Burrito   
4560      1834         1                     Chicken Salad Bowl   
4561      1834         1                     Chicken Salad Bowl   
4562      1834         1                     Chicken Salad Bowl   

                                     choice_description  choice_count   
0                                                   NaN           NaN  \
1                                          [Clementine]           1.0   
2                                               [Apple]           1.0   
3                                                   NaN           NaN   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...           5.0   
...                                                 ...           ...   
4558  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...           7.0   
4559  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...           6.0   
4560  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...           5.0   
4561  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...           3.0   
4562  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...           4.0   

      item_price  order_count  item_code  
0           2.39          110         24  
1           3.39           19         34  
2           3.39           27         35  
3           2.39           31         31  
4          16.98          717         17  
...          ...          ...        ...  
4558       11.75          365         39  
4559       11.75          365         39  
4560       11.25          110         21  
4561        8.75          110         21  
4562        8.75          110         21  

[4563 rows x 8 columns]

In [137]:
del merged_df['order_count']
merged_df.head()

order_id  quantity                              item_name   
0         1         1           Chips and Fresh Tomato Salsa  \
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   

                                  choice_description  choice_count   
0                                                NaN           NaN  \
1                                       [Clementine]           1.0   
2                                            [Apple]           1.0   
3                                                NaN           NaN   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...           5.0   

   item_price  item_code  
0        2.39         24  
1        3.39         34  
2        3.39         35  
3        2.39         31  
4       16.98         17

Moving `item_code` next to `item_name`

In [144]:
item_code_column = merged_df.pop('item_code')

In [145]:
merged_df.insert(merged_df.columns.get_loc('item_name') + 1, 'item_code', item_code_column)

merged_df

order_id  quantity                              item_name  item_code   
0            1         1           Chips and Fresh Tomato Salsa         24  \
1            1         1                                   Izze         34   
2            1         1                       Nantucket Nectar         35   
3            1         1  Chips and Tomatillo-Green Chili Salsa         31   
4            2         2                           Chicken Bowl         17   
...        ...       ...                                    ...        ...   
4558      1833         1                          Steak Burrito         39   
4559      1833         1                          Steak Burrito         39   
4560      1834         1                     Chicken Salad Bowl         21   
4561      1834         1                     Chicken Salad Bowl         21   
4562      1834         1                     Chicken Salad Bowl         21   

                                     choice_description  choice_count   
0                                                   NaN           NaN  \
1                                          [Clementine]           1.0   
2                                               [Apple]           1.0   
3                                                   NaN           NaN   
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...           5.0   
...                                                 ...           ...   
4558  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...           7.0   
4559  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...           6.0   
4560  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...           5.0   
4561  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...           3.0   
4562  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...           4.0   

      item_price  
0           2.39  
1           3.39  
2           3.39  
3           2.39  
4          16.98  
...          ...  
4558       11.75  
4559       11.75  
4560       11.25  
4561        8.75  
4562        8.75  

[4563 rows x 7 columns]

# Basic Analysis on the Cleaned Dataset

In [147]:
chipotle = merged_df.copy()

### Order ID with the highest price

In [151]:
orderid_df.sort_values(by = 'item_price', ascending = False).head()

order_id  quantity  choice_count  item_price
925        926        21          98.0      187.75
1442      1443        35          18.0      160.74
1482      1483        14          74.0      139.00
690        691        11          62.0      118.25
1785      1786        20          50.0      114.30

Fetching the same rows from the `chipotle` dataframe

In [154]:
highest_orders = [926, 1443, 1483, 691, 1786]
chipotle[chipotle['order_id'].isin(highest_orders)]

        #can be stored in a variable to use later if needed

order_id  quantity            item_name  item_code   
1684       691         1      Chicken Burrito         18  \
1685       691         1   Chicken Salad Bowl         21   
1686       691         1        Steak Burrito         39   
1687       691         1      Chicken Burrito         18   
1688       691         1         Chicken Bowl         17   
...        ...       ...                  ...        ...   
4431      1786         1     Carnitas Burrito         12   
4432      1786         1         Chicken Bowl         17   
4433      1786         4  Chips and Guacamole         25   
4434      1786         4    Canned Soft Drink         10   
4435      1786         4    Canned Soft Drink         10   

                                     choice_description  choice_count   
1684  [Tomatillo Red Chili Salsa, [Rice, Cheese, Sou...           4.0  \
1685  [Roasted Chili Corn Salsa, [Fajita Vegetables,...           5.0   
1686  [Tomatillo Red Chili Salsa, [Rice, Black Beans...           5.0   
1687  [Tomatillo Red Chili Salsa, [Rice, Pinto Beans...           5.0   
1688  [Fresh Tomato Salsa, [Rice, Black Beans, Chees...           6.0   
...                                                 ...           ...   
4431  [Fresh Tomato Salsa, [Rice, Black Beans, Guaca...           5.0   
4432  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...           5.0   
4433                                                NaN           NaN   
4434                                             [Coke]           1.0   
4435                                           [Sprite]           1.0   

      item_price  
1684        8.75  
1685       11.25  
1686        9.25  
1687        8.75  
1688       11.25  
...          ...  
4431       11.75  
4432        8.75  
4433       17.80  
4434        5.00  
4435        5.00  

[64 rows x 7 columns]

### Highest Quantity Ordered (individual)

In [163]:
chipotle.sort_values(by = 'quantity', ascending = False).head()

order_id  quantity                     item_name  item_code   
3554      1443        15  Chips and Fresh Tomato Salsa         24  \
4102      1660        10                 Bottled Water          6   
3839      1559         8                 Side of Chips         37   
3555      1443         7                 Bottled Water          6   
2411       970         5                 Bottled Water          6   

     choice_description  choice_count  item_price  
3554                NaN           NaN       44.25  
4102                NaN           NaN       15.00  
3839                NaN           NaN       13.52  
3555                NaN           NaN       10.50  
2411                NaN           NaN        7.50

### Order ID with the highest quantity (agg)

In [162]:
orderid_df.sort_values(by = 'quantity', ascending = False).head()

order_id  quantity  choice_count  item_price
1442      1443        35          18.0      160.74
925        926        21          98.0      187.75
1785      1786        20          50.0      114.30
758        759        18          34.0       86.30
1659      1660        18          14.0       65.80

Fetching the same rows from the `chipotle` dataframe

In [166]:
highest_qty = [ 1443, 926, 1786, 759, 1660]
chipotle[chipotle['order_id'].isin(highest_qty)]

        #can be stored in a variable to use later if needed

order_id  quantity                            item_name  item_code   
1847       759         1                         Chicken Bowl         17  \
1848       759         1                        Carnitas Bowl         11   
1849       759         1                   Chicken Soft Tacos         22   
1850       759         1                          Veggie Bowl         44   
1851       759         1                         Chicken Bowl         17   
1852       759         1                         Chicken Bowl         17   
1853       759         2                    Canned Soft Drink         10   
1854       759         2                    Canned Soft Drink         10   
1855       759         4                        Bottled Water          6   
1856       759         2                  Chips and Guacamole         25   
1857       759         2         Chips and Fresh Tomato Salsa         24   
2277       926         1                        Steak Burrito         39   
2278       926         1                         Chicken Bowl         17   
2279       926         1                         Chicken Bowl         17   
2280       926         1                           Steak Bowl         38   
2281       926         1                         Chicken Bowl         17   
2282       926         1                        Steak Burrito         39   
2283       926         1                      Chicken Burrito         18   
2284       926         1                         Chicken Bowl         17   
2285       926         1                         Chicken Bowl         17   
2286       926         1                   Chicken Salad Bowl         21   
2287       926         1                           Steak Bowl         38   
2288       926         1                      Chicken Burrito         18   
2289       926         1                           Steak Bowl         38   
2290       926         1                           Steak Bowl         38   
2291       926         1                      Chicken Burrito         18   
2292       926         1                         Chicken Bowl         17   
2293       926         1                         Chicken Bowl         17   
2294       926         1                     Barbacoa Burrito          2   
2295       926         1                      Chicken Burrito         18   
2296       926         1                           Steak Bowl         38   
2297       926         1                          Veggie Bowl         44   
3554      1443        15         Chips and Fresh Tomato Salsa         24   
3555      1443         7                        Bottled Water          6   
3556      1443         1                    6 Pack Soft Drink          0   
3557      1443         3                       Veggie Burrito         45   
3558      1443         4                      Chicken Burrito         18   
3559      1443         3                        Steak Burrito         39   
3560      1443         2                        Bottled Water          6   
4098      1660         2                      Chicken Burrito         18   
4099      1660         1                                Chips         23   
4100      1660         1                    Canned Soft Drink         10   
4101      1660         1                         Chicken Bowl         17   
4102      1660        10                        Bottled Water          6   
4103      1660         1                          Veggie Bowl         44   
4104      1660         1                  Chips and Guacamole         25   
4105      1660         1  Chips and Tomatillo Red Chili Salsa         30   
4425      1786         1                         Chicken Bowl         17   
4426      1786         1                     Carnitas Burrito         12   
4427      1786         1                         Chicken Bowl         17   
4428      1786         1                         Chicken Bowl         17   
4429      1786         1                        Barbac

### Highest Selling Items

In [152]:
item_df.sort_values(by = 'order_count', ascending = False).head()

item_name  order_count  item_code
17         Chicken Bowl          717         17
18      Chicken Burrito          546         18
25  Chips and Guacamole          474         25
39        Steak Burrito          365         39
10    Canned Soft Drink          290         10

### Lowest Selling Items

In [161]:
item_df.sort_values(by = 'order_count', ascending = True).head(7)

item_name  order_count  item_code
14                     Carnitas Salad            1         14
46                Veggie Crispy Tacos            1         46
26  Chips and Mild Fresh Tomato Salsa            1         26
33                       Crispy Tacos            2         33
7                                Bowl            2          7
36                              Salad            2         36
41                        Steak Salad            4         41

### Highest choice_counts of toppings and add-ons

In [155]:
chipotle.sort_values(by = 'choice_count', ascending = False).head()

order_id  quantity        item_name  item_code   
2015       820         1  Chicken Burrito         18  \
3614      1463         1      Veggie Bowl         44   
1305       538         1  Chicken Burrito         18   
2132       871         1  Chicken Burrito         18   
1274       525         1    Steak Burrito         39   

                                     choice_description  choice_count   
2015  [[Fresh Tomato Salsa (Mild), Tomatillo-Green C...          10.0  \
3614  [[Fresh Tomato Salsa (Mild), Tomatillo-Green C...          10.0   
1305  [[Tomatillo-Green Chili Salsa (Medium), Roaste...          10.0   
2132  [[Tomatillo-Green Chili Salsa (Medium), Roaste...          10.0   
1274  [[Roasted Chili Corn Salsa (Medium), Fresh Tom...           9.0   

      item_price  
2015        8.49  
3614        8.49  
1305       10.98  
2132       10.98  
1274        8.99

The above were some basic analysis examples done on the dataset to draw insights. <br> <br>
The analysis can be taken a step further by employing visualizations. 

# Exporting dataset to CSV

In [167]:
chipotle.to_csv('cleaned_chipotle.csv', index=False)

In [168]:
item_df.to_csv('items_chipotle.csv', index=False)

In [169]:
orderid_df.to_csv('orders_chipotle.csv', index=False)